### DISCUSSION TITLE
Possible for Huawei Watch GT4 (HarmonyOS) to display glucose level?

### DISCUSSION THREAD

**Discussion by bjcryss** [NightscoutFoundation/xDrip#3350] [Q&A]
Hey there, I know some guys tried sideloading xdrip+ onto Huawei watch 3, and it's kind of obvious this would not work properly.

I'm not looking to have xdrip run on the watch, but would be really nice if you know of some Huawei app that can pick-up data from xdrip+/nightscout and display it on the watch ? Or maybe an app meant for Android or Wear OS but which can be sideloaded into HarmonyOS?

For starters, just to have the live glicemic value appear on the watch would suffice. If not loaded directly on the watch face, but maybe loaded into an app panel for easier access. 

I haven't tried using a browser app and opening nightscout, but will try that and see how it goes if it's possible.

Any idea appreciated !

**Comment by PhilTU** [comment]
WatchGlucose for Wear OS I think interrogates LibreLinkUp but check (Libre only, obv)

G-Watch Wear is for WearOS and receives from xDrip+

https://play.google.com/store/apps/details?id=com.silentavalanche.blose&hl=en&gl=US is another Linkup option

**Reply by bjcryss** [reply]
Thanks for the ideas, but GT4 is running HarmonyOS, not really Wear OS, so i have no idea how those apps would pan out.

Scratch that, just found out that GT4 is actually running LiteOS ! which is a really dumbed down version of HarmonyOS, and is very very limited in apps that you can install on it. BURN !! bought the damn GT4 watch with hope (middlefinger)
Only "Huawei Watch" series  which costs 3 times more than GT series actually run HarmonyOS which can support proper apps

**Comment by olo3345** [comment]
how do you sideload apps on the GT4 in the first place ?

**Reply by bjcryss** [reply]
GT4 is actually running LiteOS ! which is a really dumbed down version of HarmonyOS, and is very very limited in apps that you can install on it. so i think sideloading is out of the question

**Comment by thejazzs** [comment]
Hello.
I am using Huawei watch GT6 and searched almost whole internet to find out a way about seeing the glucose levels at the watch. 
None of the solutions were for Huawei (Harmony OS) unfortunately.
I've checked Xdrip+ , Juggluco, Gluco Datahandler, etc. as 3rd parties, but none of them worked.
Please kindly share some usefull information If anyone knows how to solve it.
It will be highly appreciated in case.
Thank you.
